lms/static/js/dialog_tab_controls.js

```diff
@@ -0,0 +1,76 @@
+var DialogTabControls = (function() {
+    'use strict';
+
+    var focusableChildren,
+        numElements,
+        currentIndex,
+        setCurrentIndex = function(currentElement) {
+            var elementIndex = focusableChildren.index(currentElement);
+            if (elementIndex >= 0) {
+                currentIndex = elementIndex;
+            }
+        },
+        initializeTabKeyValues = function(elementName, $closeButton) {
+            focusableChildren = $(elementName).find(
+                'a, input[type=text], input[type=submit], select, textarea, button'
+            );
+            if ($closeButton) {
+                focusableChildren = focusableChildren.add($closeButton);
+            }
+            numElements = focusableChildren.length;
+            currentIndex = 0;
+            focusableChildren[currentIndex].focus();
+            focusableChildren.on('click', function() {
+                setCurrentIndex(this);
+            });
+        },
+        focusElement = function() {
+            var focusableElement = focusableChildren[currentIndex];
+            if (focusableElement) {
+                focusableElement.focus();
+            }
+        },
+        focusPrevious = function() {
+            currentIndex--;
+            if (currentIndex < 0) {
+                currentIndex = numElements - 1;
+            }
+
+            focusElement();
+
+            return false;
+        },
+        focusNext = function() {
+            currentIndex++;
+            if (currentIndex >= numElements) {
+                currentIndex = 0;
+            }
+
+            focusElement();
+
+            return false;
+        },
+        setKeydownListener = function($modal, $closeButton) {
```

Comment:
prefer a more generic variable name like the method above.  If this will work for non-modal elements we should not prescribe the usage with a name like this